src/t_string.c

```diff
@@ -207,83 +212,57 @@ static int getExpireMillisecondsOrReply(client *c, robj *expire, int flags, int
  * EX/EXAT/PX/PXAT arguments. Unit is updated to millisecond if PX/PXAT is set.
  */
 int parseExtendedStringArgumentsOrReply(client *c, int *flags, int *unit, robj **expire, int command_type) {
-
     int j = command_type == COMMAND_GET ? 2 : 3;
     for (; j < c->argc; j++) {
         char *opt = c->argv[j]->ptr;
-        robj *next = (j == c->argc-1) ? NULL : c->argv[j+1];
+        robj *next = (j == c->argc - 1) ? NULL : c->argv[j + 1];
 
-        if ((opt[0] == 'n' || opt[0] == 'N') &&
-            (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
-            !(*flags & OBJ_SET_XX) && (command_type == COMMAND_SET))
-        {
+        if ((opt[0] == 'n' || opt[0] == 'N') && (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
+            !(*flags & OBJ_SET_XX) && (command_type == COMMAND_SET)) {
             *flags |= OBJ_SET_NX;
-        } else if ((opt[0] == 'x' || opt[0] == 'X') &&
-                   (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
-                   !(*flags & OBJ_SET_NX) && (command_type == COMMAND_SET))
-        {
+        } else if ((opt[0] == 'x' || opt[0] == 'X') && (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
+                   !(*flags & OBJ_SET_NX) && (command_type == COMMAND_SET)) {
             *flags |= OBJ_SET_XX;
-        } else if ((opt[0] == 'g' || opt[0] == 'G') &&
-                   (opt[1] == 'e' || opt[1] == 'E') &&
-                   (opt[2] == 't' || opt[2] == 'T') && opt[3] == '\0' &&
-                   (command_type == COMMAND_SET))
-        {
+        } else if ((opt[0] == 'g' || opt[0] == 'G') && (opt[1] == 'e' || opt[1] == 'E') &&
+                   (opt[2] == 't' || opt[2] == 'T') && opt[3] == '\0' && (command_type == COMMAND_SET)) {
```

Comment:
These are nice to read vertically as "get" vs "GET". I want to disable formatting for this function.

Reply:
I'm curious.  Does this code really run that much faster than strcasecmp?  If it does, can we refactor it into a macro or an inline or something so that it reads something like the following?
~~~
} else if (IS_GET(opt) && (command_type == COMMAND_SET))
        {
~~~

Reply:
I don't know but i don't want to spend time on it now. Feel free to improve this if you want.

Reply:
The conversion of a char to upper is just one bit flip, `c & ~0x20`.